WalletWasabi/WabiSabi/Client/WabiSabiHttpApiClient.cs

```diff
@@ -101,56 +99,23 @@ private async Task<HttpResponseMessage> SendWithRetriesAsync(RemoteAction action
 			catch (HttpRequestException e)
 			{
 				Logger.LogTrace($"Attempt {attempt} to perform '{action}' failed with {nameof(HttpRequestException)}: {e.Message}.");
-				AddException(exceptions, e);
 			}
 			catch (OperationCanceledException e)
 			{
 				Logger.LogTrace($"Attempt {attempt} to perform '{action}' failed with {nameof(OperationCanceledException)}: {e.Message}.");
-				AddException(exceptions, e);
 			}
 			catch (Exception e)
 			{
 				Logger.LogDebug($"Attempt {attempt} to perform '{action}' failed with exception {e}.");
-
-				if (exceptions.Any())
-				{
-					AddException(exceptions, e);
-					throw new AggregateException(exceptions.Keys);
-				}
-
 				throw;
 			}
 
-			try
-			{
-				// Wait before the next try.
-				await Task.Delay(250, combinedToken).ConfigureAwait(false);
-			}
-			catch (Exception e)
-			{
-				AddException(exceptions, e);
-			}
+			// Wait before the next try.
+			await Task.Delay(250, combinedToken).ConfigureAwait(false);
 
 			attempt++;
 		}
-		while (!combinedToken.IsCancellationRequested);
-
-		throw new AggregateException(exceptions.Keys);
-	}
-
-	private static void AddException(Dictionary<Exception, int> exceptions, Exception e)
-	{
-		bool Predicate(KeyValuePair<Exception, int> x) => e.GetType() == x.Key.GetType() && e.Message == x.Key.Message;
-
-		if (exceptions.Any(Predicate))
-		{
-			var first = exceptions.First(Predicate);
-			exceptions[first.Key]++;
-		}
-		else
-		{
-			exceptions.Add(e, 1);
-		}
+		while (true);
```

Comment:
What was wrong with 

```cs
while (!combinedToken.IsCancellationRequested);
``` 
?

Isn't `while (true);` a bit dangerous?

We could use the `attempt` counter to have an absolute maximum tries. 
For example 10.

Reply:
> What was wrong with

The method will just return without throwing an exception.

> We could use the attempt counter to have an absolute maximum tries.

We do not need to, the max attempts are determined by the timeout.